### PR TITLE
Occupation classifier

### PR DESCRIPTION
**Pull request by tweddielin**
This PR is mainly just a new module occupation_classifiers in algorithms. Now it has only one class SocClassifier() which is a SOC code classifier implemented by gensim.

### CODE REVIEW

File: skills_ml/algorithms/occupation_classifiers/classifiers.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    Example:
+
+    from airflow.hooks import S3Hook
+    from skills_ml.algorithms.occupation_classifiers.classifiers import SocClassifier
+
+    s3_conn = S3Hook().get_conn()
+    Soc = SocClassifier(s3_conn=s3_conn)
+    predicted_soc = Soc.classify(jobposting, mode='top')
+    """
+    def __init__(self, model_name=MODEL_NAME, s3_path=PATHTOMODEL, s3_conn=None):
```
Comment (thcrock) [inline skills_ml/algorithms/occupation_classifiers/classifiers.py]: I would actually rather there not be a default s3 path. That is specific to our infrastructure. It is easier to change infrastructure if we only control these paths from a config file that doesn't get checked into a repository.

File: skills_ml/algorithms/occupation_classifiers/classifiers.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+from gensim.models import Doc2Vec
+
+from skills_ml.algorithms.string_cleaners import NLPTransforms
+
+from skills_utils.s3 import download
+
+PATHTOMODEL = 'open-skills-private/model_cache/'
+LOOKUP = 'lookup_va_0605.json'
+
+MODEL_NAME = 'gensim_doc2vec_va_0605'
```
Comment (thcrock) [inline skills_ml/algorithms/occupation_classifiers/classifiers.py]: I think we should document a bit about how this pretraining happened. What dataset, what time period, how many examples. Is there any of that metadata still easily available?

File: skills_ml/algorithms/occupation_classifiers/classifiers.py [hunk 3]
```diff
@@ -2,22 +2,32 @@
...
 from gensim.models import Doc2Vec
 
 from skills_ml.algorithms.string_cleaners import NLPTransforms
 
-from skills_utils.s3 import download
+from skills_utils.s3 import download, split_s3_path
 
-PATHTOMODEL = 'open-skills-private/model_cache/'
-LOOKUP = 'lookup_va_0605.json'
+with open('config.yaml') as f:
```
Comment (thcrock) [inline skills_ml/algorithms/occupation_classifiers/classifiers.py]: I actually don't think we want to introduce any config parsing in skills-ml. It's going to be cleaner to handle the config in one place (skills-airflow) and pass in those variables to skills-ml objects.

So the model cache s3 prefix can just be an argument sent to the constructor.

Comment (tweddielin) [inline skills_ml/algorithms/occupation_classifiers/classifiers.py]: got it.

**Pull request merged**